### PR TITLE
Make athene2-editor a git submodule

### LINKED ISSUE #74: Merge athene2 and athene2-editor
The workflow for doing editor changes is really bad. Should we merge athene2-editor into this repository?

cc @creitiv @inyono @jkempff

### PR DESCRIPTION
**Pull request by inyono**
Closes #74.

@arekkas @creitiv @Knorrke. I'll documented the new workflow for the editor below. Do you consider this an improvement over the old way of doing things? IMHO only the testing improved, since the changes `athene2-editor` can be directly seen in the usual local environment. But committing changes still needs two pull requests (for those that can't directly push to `athene2`).

# Workflow

**tl;dr**. Basically, the directory `src/asssets/athene2-editor` will be treated as a own git repository and the `athene2` repository just refers to the commit to use. So to make changes, we both have to commit changes to the `athene2-editor` repository and update the reference in `athene2` to the new commit. Furthermore, cloning and pulling must include the submodule, too.

## Cloning

Add `--resursive` to clone submodules, too.

```
git clone git@github.com:serlo-org/athene2.git --recursive
```

## Pulling external changes (for athene2 and athene2-editor)

As usual, pull athene2. Also update submodules

```
git submodule update --init --recursive
```

**Note:** *This does not pull the latest commit of athene2-editor, but the proper commit specified in the athene2 repository. If you want to update the version of athene2-editor, follow the directions below.*

## Committing changes to athene2-editor

Either change the submodule in athene2 or work with the athene2-editor repository directly.

### Make changes to the submodule

**tl;dr**. Commit the changes in `src/assets/athene2-editor` and push resp. submit pull requests as usual.

First, commit the changes in athene2-editor:

```
cd src/assets/athene2-editor
git checkout -b BRANCH
git add .
git commit
git push -u REMOTE BRANCH
```

Now you can submit a pull request to athene2-editor.

### Directly modify athene2-editor

As usual, make changes to athene2-editor, and submit a pull request to athene2-editor.  

## Updating the editor in the athene2 repository

**tl;dr**. Pull the changes in `src/assets/athene2-editor` and commit `src/assets/athene2-editor` (will be treaded as a file by the `athene2` directory and contains the reference to the `athene2-editor` commit)

To update the editor in athene2 (for example, after a pull request got merged), follow these steps:

```
git checkout -b BRANCH
cd src/assets/athene2-editor
git pull origin master
cd ..
git add athene2-editor
git commit
git push -u REMOTE BRANCH
```

Finally submit a pull request to athene2.

### CODE REVIEW

**Comment by arekkas** [comment]
This looks :tada:

File: bin/deploy.sh [hunk 1]
```diff
@@ -21,14 +21,14 @@
...
 
 #
-# OS Test 
-# $1 = OS 
+# OS Test
+# $1 = OS
 # Usage: if is_os 'darwin'; then
 #
 git status
 git pull -ff
```
Comment (arekkas) [inline bin/deploy.sh]: Do we need to do some recursive stuff here as well?

Comment (inyono) [inline bin/deploy.sh]: Ah yeah, you're right. Is this the only script used for deployment?

Comment (arekkas) [inline bin/deploy.sh]: yup

File: bin/deploy.sh [hunk 2]
```diff
@@ -21,14 +21,15 @@
...
 
 #
-# OS Test 
-# $1 = OS 
+# OS Test
+# $1 = OS
 # Usage: if is_os 'darwin'; then
 #
 git status
-git pull -ff
```
Comment (inyono) [inline bin/deploy.sh]: @arekkas Removed `-ff` (since the corresponding parameter is `--ff` and the default behaviour anyway). Added submodule updating to deploy script

**Comment by arekkas** [comment]
Any idea why tests fail?

https://travis-ci.org/serlo-org/athene2/jobs/70283715

**Comment by inyono** [comment]
Now it works, changed the git submodule url to HTTPS from SSH. Seems like Travis was not correctly configured for this, I guess.

**Comment by arekkas** [comment]
Did you do a recursive search over all files with `athene2-editor` and replaced the paths? If yes, this looks good for me to merge

**Comment by inyono** [comment]
Yeah, I did.

**Comment by arekkas** [comment]
Ok great :)
I'll postpone this, because I want to extensively test if this works so we don't break production.

**Comment by arekkas** [comment]
Sorry for not having merged this yet. I'll get on it soon, promised.

**Comment by arekkas** [comment]
First time pull with: `git submodule update --init`

**Comment by arekkas** [comment]
:balloon:
